### ISSUE TITLE
Preventing automatic link generation with ! fails with pypy interepreter

### ISSUE THREAD

**Issue by CendioMartin** [moinwiki/moin-1.9#92]
As per @ThomasWaldmann suggestion in #91, running with pypy instead of CPython is way forward until Moin2 is stable.

I stumbled upon a problem with negating automatic link generation. Prepending !CamelCase should produce 
`CamelCase`
but does not. It's successful in not tuning it to a link, but it ouputs as
`!CamelCase`

It's easy enough to reproduce:
Download the latest release of pypy2:
`curl -O https://downloads.python.org/pypy/pypy2.7-v7.3.9-linux64.tar.bz2`

Clone moin repo
`git clone https://github.com/moinwiki/moin-1.9`

Start Desktop Edition of moinmoin
`/path/to/pypy wikiserver.py`

Point your browser to the wiki and create a page with a link and watch the output.
(I had to edit wikiserverconfig.py and insert `DesktopEdition = True` to be able to edit/create pages)

Press Ctrl+C to stop the wikiserver.py and start it with CPython:
`python wikiserver.py`

Refresh that same page and you should see it output as expected.

**Comment by ThomasWaldmann** [comment]
did you try cleaning the cache when switching interpreters?

```
moin ... maint cleancache
```

or just kill the `cache` subdir of the respective page directory inside the `data_dir`?

**Comment by CendioMartin** [comment]
Yes, I've tried those steps at some point in my journey to troubleshoot this issue in our production wiki.
This is reproducable with a new install/DesktopEdition (as per my steps outlined above) as well.
